src/components/ecommerce/ecommerceInput.component.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { ThemedComponentProps } from '@kitten/theme';
+import {
+  Input,
+  InputProps,
+} from '../common/input.component';
+
+interface ComponentProps {
+  formatter: (value: string, stateValue: string) => string;
+  validator: (value: string) => boolean;
```

Comment:
can we use `ValidateInput` with it's regex prop for this purpose?

Reply:
Don't like this input at all. It forces me to create another one statefull component.